packages/extensions/src/shared/utils/define-extension.ts

```diff
@@ -10,11 +11,19 @@ import type {
 	PanelConfig,
 } from '../types/index.js';
 
-export function defineInterface<T extends InterfaceConfig>(config: T): T {
+type CustomConfig<T extends object> = { [K in string]: K extends keyof T ? never : unknown };
```

Comment:
`extends object` is unnecessary since the subsequent definition already narrows it down to a plain object

```suggestion
type CustomConfig<T> = { [K in string]: K extends keyof T ? never : unknown };
```